src/Audis.Analyzer.Common/Audis.Analyzer.Common/Extensions/V1/IListExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using Audis.Analyzer.Contract.V1;
+using Audis.Primitives;
+
+namespace Audis.Analyzer.Common.Extensions.V1
+{
+    public static class IListExtensions
+    {
+        public static void Add(
+            this IList<SuggestedKnowledgeDto> suggestedKnowledges,
+            string identifier,
+            string value,
+            double probability = 1.0,
+            DateTime? lastUpdated = null,
+            KnowledgeOrigin origin = KnowledgeOrigin.Client)
+        {
+            if (value != null)
+            {
+                suggestedKnowledges.Add(new SuggestedKnowledgeDto
```

Comment:
Hm, was tun wir, wenn es im SuggestedKnowledge schon ein Knowledge mit dem übergebenen Identifier gibt? Grundsätzlich wäre es für mich schlüssiger, dann das `Values`-Array zu erweitern anstatt ein neues `SuggestedKnowledge` zu erstellen. Ich weiß aber nicht, wie wir dann mit `Probability` umgehen. `LastUpdate` und `Origin` können wir mit den neuen Werten aktualisieren.

@astelzhammer Hast du dazu auch eine Meinung?

Reply:
Ich würde auf alle Fälle die `Values`-Arrays miteinander mergen. Bezüglich `Probability` sollte vermutlich eine neue Wahrscheinlichkeit berechnet werden (Addieren/Durchschnitt/...).

Reply:
Gute Idee @meinsiedler 
Und bei mehreren Values muss dann ja auch eine `AnswerId `gesetzt sein oder? Zumindest haben mehrere Values immer zu einer Exception geführt ohne AnswerIds.

Falls das zuviele Fragezeichen aufwirft, könnte man den Wert entweder ersetzen oder auf basis von der Probability das wahrscheinlichere übernehmen?

Reply:
@astelzhammer die Durchschnitts-Berechnung führt da halt dazu, dass ab dem 3.Mal zusammenführen nicht mehr der Durchschnitt über alle Elemente gebildet wird, da ja der Durchschnitt mit dem zuvor berechneten Durchschnitt berechnet wird - ist das trotzdem in Ordnung?
`((0.5 + 0.75) / 2 + 0.6) / 2`